services/dedup/types/types.go

```diff
@@ -1,9 +1,39 @@
 package types
 
+import (
+	"errors"
+)
+
 // BatchKey represents a key in a batch
 type BatchKey struct {
 	// Index is the index of the key in the batch (used for discriminating between keys with the same value)
 	Index int
 	// Key is the value of the key
 	Key string
 }
+
+// DB is the interface that all dedup database implementations must satisfy
+type DB interface {
+	// Get returns a map of keys that exist in the database
+	Get(keys []string) (map[string]bool, error)
+	// Set stores keys in the database with a TTL
+	Set(keys []string) error
+	// Close closes the database connection
+	Close()
+}
+
+// DedupInterface is the interface for deduplication service
+type DedupInterface interface {
```

Comment:
Keep interface names clean, use Impl if there is a naming clash with the implementation struct and you need to keep the implementation struct exported
```suggestion
type Dedup interface {
```